node/src/effect.rs

```diff
@@ -1555,6 +1555,25 @@ impl<REv> EffectBuilder<REv> {
         .into_iter()
         .collect()
     }
+
+    /// Collects booking blocks for given eras.
+    /// Booking block is a switch block.
+    /// It is expected that the caller of this function knows in which era the booking blocks live.
+    pub(crate) async fn collect_booking_blocks<I: IntoIterator<Item = EraId>>(
+        self,
+        booking_blocks_era_ids: I,
+    ) -> Option<HashMap<EraId, BlockHeader>>
+    where
+        REv: From<StorageRequest>,
+    {
+        futures::future::join_all(booking_blocks_era_ids.into_iter().map(|era_id| {
+            self.get_key_block_for_era_id_from_storage(era_id)
```

Comment:
That gets the switch block in `era_id - 1` — is that intended?

Reply:
No. It will get the switch block in `era_id`.

Reply:
But the "key block for era" is the switch block in the previous era… maybe that method should have a different name then.

Reply:
Wait, it will get the switch block in era `era_id - 1` :tired_face:

Reply:
Actually, I think I wanted to read switch blocks (not key blocks) and just used a wrong method.

Reply:
Fixed in [`ac8d609` (#1068)](https://github.com/CasperLabs/casper-node/pull/1068/commits/ac8d609c18a52d4da666b399b6ec44360deb98d6) and then fixed again in [`42efc26` (#1068)](https://github.com/CasperLabs/casper-node/pull/1068/commits/42efc26740f2029c1fb7a3268f07abfaa312226e)